lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,208 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { ApiFile } from '@cumulus/types/api/files';
+import {
+  CollectionPgModel,
+  GranulePgModel,
+  PostgresFile,
+  PostgresGranuleRecord,
+  tableNames,
+  translateApiGranuleToPostgresGranule,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/granules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+const { getBucket, getKey } = require('@cumulus/api/lib/FileUtils');
+const { deconstructCollectionId } = require('@cumulus/api/lib/utils');
+
+export interface GranulesAndFilesMigrationSummary {
+  granulesSummary: MigrationSummary,
+  filesSummary: MigrationSummary,
+}
+
+/**
+ * Migrate granules record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} record
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<void>}
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateGranuleRecord = async (
+  record: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(record, schemas.granule);
+  const { name, version } = deconstructCollectionId(record.collectionId);
+  const collectionPgModel = new CollectionPgModel();
+
+  const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+    knex,
+    { name, version }
+  );
+
+  const existingRecord = await knex<PostgresGranuleRecord>('granules')
+    .where({ granule_id: record.granuleId, collection_cumulus_id: collectionCumulusId })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+  }
+
+  const granule = await translateApiGranuleToPostgresGranule(record, knex, collectionCumulusId);
```

Comment:
It seems kind of inconsistent to be doing the collection lookup in this function rather than in `translateApiGranuleToPostgresGranule`. If the point of `translateApiGranuleToPostgresGranule` is that it can convert a Dynamo granule record to a PG record, then it needs to do the collection lookup itself

Reply:
By `collection lookup` are you referring to retrieving the `collectionCumulusId`? Because I still need that to check for an existing record. I thought I could just pass it into the `translate` function instead of having to run the same lookup but I can see how that isn't generalizable if someone else uses the `translate` function. 

I'd still need to look up `collectionCumulusId` though. @markdboyd

Reply:
[`c58cf1b` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/c58cf1bbdd841e01b1621ee0b002c6a4c0b8d210)